classes/class-author.php

```diff
@@ -35,31 +50,38 @@ public function __construct( $user_id, $user_meta = array() ) {
 	/**
 	 * Get various user meta data
 	 *
-	 * @param string $name
+	 * @todo Make sure this is being covered in the unit tests.
+	 *
+	 * @param string $name User meta key.
 	 *
-	 * @throws \Exception
+	 * @throws \Exception Meta not found | User not found.
 	 *
 	 * @return string
 	 */
 	public function __get( $name ) {
-		if ( 'display_name' === $name ) {
-			return $this->get_display_name();
-		} elseif ( 'avatar_img' === $name ) {
-			return $this->get_avatar_img();
-		} elseif ( 'avatar_src' === $name ) {
-			return $this->get_avatar_src();
-		} elseif ( 'role' === $name ) {
-			return $this->get_role();
-		} elseif ( 'agent' === $name ) {
-			return $this->get_agent();
-		} elseif ( ! empty( $this->user ) && 0 !== $this->user->ID ) {
-			return $this->user->$name;
-		} else {
-			throw new \Exception( "Unrecognized magic '$name'" );
+		switch ( $name ) {
+			case 'display_name':
+			case 'avatar_img':
+			case 'avatar_src':
+			case 'role':
+			case 'agent':
+				$getter = "get_{$name}";
+				return $this->$getter();
+			default:
```

Comment:
Noting this code change.

Reply:
I'm still not sure if this is tested or not.